ctapipe/io/containers.py

```diff
@@ -78,6 +77,7 @@ class RawCameraContainer(Container):
                             "integrated ADC data (n_pixels)"))
     adc_samples = Item(Map(), ("map of channel to arrays of "
                                "(n_pixels, n_samples)"))
+    num_samples = Item(int, "number of time samples")
```

Comment:
it's strange that it changes per event in pyhessio, but that must have to do with how the data is read. That is still header info, not event info.  Why is it a problem reading it from the inst container?

Of course remember that in the end reading any camera or layout info from the hessio containers will disappear, so this is all just a temporary hack. Eventually this will be an interface to a database or local data description file handled by the instrument module. You'll get a structure somewhat like `CameraGeometry` that contains a num_channels field. 

And of course even further down the line, there will be no channels visible to the calibration code, since we will start with DL0 data and they are dropped by the camera in R1.